types/lodash/index.d.ts

```diff
@@ -12465,71 +12465,69 @@ declare namespace _ {
          * @param defaultValue The value returned if the resolved value is undefined.
          * @return Returns the resolved value.
          */
-        get<TResult>(
-            object: any,
-            path: Many<StringRepresentable>,
-            defaultValue: TResult
-        ): TResult;
+        // when object is empty - definitely default value will be returned
+        get<TDefault>(
+            object: null | undefined,
+            path: Many<StringRepresentable>, // ** Many<StringRepresentable> instead of any
+            defaultValue: TDefault // ** no longer optional
+        ): TDefault;
+
+        // ** same as above, but with no default value
+        get(
+            object: null | undefined,
+            path: Many<StringRepresentable>
+        ): undefined;
 
-        /**
-         * @see _.get
-         */
-        get<TObject extends object, TKey extends keyof TObject>(
-            object: TObject | null | undefined,
+        // when depth of object is 1 and path to value is right - or value, or default will be returned (in case value is undefined)
+        get<TObject extends object, TKey extends keyof TObject, TDefault>(
+            object: TObject,
             path: TKey | [TKey],
-            defaultValue: TObject[TKey]
-        ): TObject[TKey];
+            defaultValue: TDefault // ** no longer optional
+        ): TObject[TKey] | TDefault;
 
-        /**
-         * @see _.get
-         */
+        // ** same as above, but with no default value
         get<TObject extends object, TKey extends keyof TObject>(
             object: TObject,
             path: TKey | [TKey]
         ): TObject[TKey];
 
-        /**
-         * @see _.get
-         */
-        get<TObject extends object, TKey extends keyof TObject>(
+        // when object may or may not arrive
+        get<TObject extends object, TKey extends keyof TObject, TDefault>(
             object: TObject | null | undefined,
             path: TKey | [TKey],
+            defaultValue: TDefault
+        ): TObject[TKey] | TDefault;
```

Comment:
This is basically a duplicate of the last overload that I commented on. If we decide to keep it, you should delete the other one and keep this one, because it allows null/undefined objects.